### DISCUSSION TITLE
Osoyoos broke H3EK Sapien?

### DISCUSSION THREAD

**Discussion by ghost** [num0005/Osoyoos-Launcher#11] [General]
Ever since I set up Osoyoos with the H3EK, Sapien has just refused to work. Sapien starts normally with a prompt to open a scenario file, but doesn't go much further than there.

I tried multiple clean installs of H3EK and running Sapien without Osoyoos to no avail. I even went into appdata to delete the config file for Osoyoos. All with no luck. I installed the H2EK and H2 Sapien works perfectly.

Can't say with 100% confidence it was Osoyoos but as soon as I set it up in my H3EK directory, my Sapien will not function.

**Comment by General-101** [comment]
Gonna have to see your Sapien reports folder. I doubt this was Osoyoos. It doesn't do anything but launch the program.

**Comment by num0005** [comment]
That sounds like an MCC/windows issue. Did you check you are opening a Halo 3 level and not a Halo 1 or Halo 2 level? The way the file selection dialog works results in it defaulting to the path a `sapien.exe` last used. Do you get an error message to the effect of the tag being invalid or the working directory not being correct?

**Comment by ghost** [comment]
Hey there. I'm 100% sure that I'm selecting a Halo 3 level.

Here's the debug from reports_00 after I boot up Sapien. 

It shows a lot of missing tags, which is strange as they have all been extracted correctly (i.e it's not tags/tags/...). I was told by someone that I had to boot up Sapien from Osoyoos to allow real-time tag editing in Sapien with Guerrilla, so it might be unable to detect the tags?

For the record. Guerrilla finds the tags folder no problem.

File: [debug.txt](https://github.com/num0005/Osoyoos-Launcher/files/7349662/debug.txt)

**Comment by num0005** [comment]
Can you find the missing tags using Guerilla?

**Comment by num0005** [comment]
Or File Explorer?

**Comment by ghost** [comment]
Just had a look. It seems most of the tag warnings in the debug.txt file are armour permutations, not sure if that would be causing any issues. I can find the tag in both file explorer and guerrilla that the permutations reference though (mp_masterchief.model).

**Comment by ghost** [comment]
I’ll try to re-install H3EK to a different directory tomorrow and hope that it will work. 

This is all likely beyond the scope of Osoyoos and was probably a coincidence, but I appreciate the help you’ve given so far.

**Comment by num0005** [comment]
that doesn't seem to be one of the tags it was complaining about unless I missed something, did it generate a crash dump (`minidump.dmp`)?

**Comment by ghost** [comment]
Nope, tried searching for it but can't find a minidump file.

**Comment by ghost** [comment]
Tried re-installing on another hard drive, also no luck. Very weird since H2 Sapien works perfectly fine

**Comment by num0005** [comment]
Maybe check logs in event viewer? If it's a stack buffer overflow you won't get a minidump since windows terminates the process straightaway.

**Comment by num0005** [comment]
I'm going to convert this to a discussion since it's likely unrelated to the launcher.
